addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/utils/ByteArrayFileCache.java

```diff
@@ -0,0 +1,236 @@
+        } catch (IOException e) {
+            logger.warn("Could not write file '{}' to cache", fileInCache.getName(), e);
+        }
+    }
+
+    /**
+     * Checks if the key is present in the cache.
+     *
+     * @param key the key whose presence in the cache is to be tested
+     * @return true if the cache contains a file for the specified key
+     */
+    public boolean containsKey(String key) {
+        return getUniqueFile(key).exists();
+    }
+
+    /**
+     * Removes the file associated with the given key from the cache.
+     *
+     * @param key the key whose associated file is to be removed
+     */
+    public void remove(String key) {
+        deleteFile(getUniqueFile(key));
+    }
+
+    /**
+     * Deletes the given {@link File}.
+     *
+     * @param fileInCache the {@link File}
+     */
+    private void deleteFile(File fileInCache) {
+        if (fileInCache.exists()) {
+            logger.debug("Deleting file '{}' from cache", fileInCache.getName());
+            fileInCache.delete();
+        } else {
+            logger.debug("File '{}' not found in cache", fileInCache.getName());
+        }
+    }
+
+    /**
+     * Removes all files from the cache.
+     */
+    public void clear() {
+        File[] filesInCache = cacheFolder.listFiles();
+        if (filesInCache != null && filesInCache.length > 0) {
+            logger.debug("Deleting all files from cache");
+            Arrays.stream(filesInCache).forEach(File::delete);
+        }
+    }
+
+    /**
+     * Returns the content of the file associated with the given key, if it is present.
+     *
+     * @param key the key whose associated file is to be returned
+     * @return the content of the file associated with the given key
+     */
+    public byte[] get(String key) {
+        return readFile(getUniqueFile(key));
+    }
+
+    /**
+     * Reads the content from the given {@link File}, if it is present.
+     *
+     * @param fileInCache the {@link File}
+     * @return the content of the file
+     */
+    private byte[] readFile(File fileInCache) {
+        if (fileInCache.exists()) {
+            logger.debug("Reading file '{}' from cache", fileInCache.getName());
+            try {
+                return Files.readAllBytes(fileInCache.toPath());
+            } catch (IOException e) {
+                logger.warn("Could not read file '{}' from cache", fileInCache.getName(), e);
+            }
+        } else {
+            logger.debug("File '{}' not found in cache", fileInCache.getName());
+        }
+        return new byte[0];
+    }
+
+    /**
+     * Creates a unique {@link File} from the key with which the file is to be associated.
+     *
+     * @param key the key with which the file is to be associated
+     * @return unique file for the file associated with the given key
+     */
+    private File getUniqueFile(String key) {
+        // TODO: store / cache file internally for faster operations
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?